src/PageHeader/PageHeader.tsx

```diff
@@ -0,0 +1,375 @@
+  return (
+    <Box
+      sx={merge<BetterSystemStyleObject>(
+        {
+          display: isHidden ? 'none' : 'flex',
+          flexDirection: 'row',
+          alignItems: 'center',
+          gap: '0.5rem',
+          flexGrow: '1',
+          justifyContent: 'right',
+        },
+        sx,
+      )}
+    >
+      {children}
+    </Box>
+  )
+}
+
+// PageHeader.TitleArea: The main title area of the page. Visible on all viewports.
+// PageHeader.TitleArea Sub Components: PageHeader.LeadingAction, PageHeader.LeadingVisual, PageHeader.Title, PageTitle.TrailingVisual, PageHeader.TrailingAction, PageHeader.Actions
+// PageHeader.LeadingAction and PageHeader.TrailingAction are only visible on regular viewports therefore they come as visible on narrow viewports and their visibility can be managed by their exposed `visible` prop
+// ---------------------------------------------------------------------
+const TitleArea: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({children, sx = {}, hidden = false}) => {
+  const isHidden = useResponsiveValue(hidden, false)
+  return (
+    <Box
+      sx={merge<BetterSystemStyleObject>(
+        {gap: '0.5rem', display: isHidden ? 'none' : 'flex', flexDirection: 'row', alignItems: 'flex-start'},
+        sx,
+      )}
+    >
+      {children}
+    </Box>
+  )
+}
+
+const LeadingAction: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({
+  children,
+  sx = {},
+  hidden = hiddenOnNarrow,
+}) => {
+  const isHidden = useResponsiveValue(hidden, false)
+  return (
+    <Box
+      sx={merge<BetterSystemStyleObject>(
+        {display: isHidden ? 'none' : 'flex', alignItems: 'center', height: '2rem'},
+        sx,
+      )}
+    >
+      {children}
+    </Box>
+  )
+}
+
+const LeadingVisual: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({children, sx = {}, hidden = false}) => {
+  const isHidden = useResponsiveValue(hidden, false)
+  return (
+    <Box
+      sx={merge<BetterSystemStyleObject>(
+        {
+          display: isHidden ? 'none' : 'flex',
+          alignItems: 'center',
+          height: '2rem',
+        },
+        sx,
+      )}
+    >
+      {children}
+    </Box>
+  )
+}
+
+export type TitleProps = {
+  variant?: 'subtitle' | 'medium' | 'large' | ResponsiveValue<'subtitle' | 'medium' | 'large'>
+  // Check if we need responsive values for heading is so should we update as prop's type for Heading component?
+  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
+} & PageHeaderProps
+
+const Title: React.FC<React.PropsWithChildren<TitleProps>> = ({
+  children,
+  sx = {},
+  variant = 'medium',
+  hidden = false,
+  as = 'h3',
+}) => {
+  const currentVariant = useResponsiveValue(variant, 'medium')
+  const isHidden = useResponsiveValue(hidden, false)
+  return (
+    <Heading
+      as={as}
+      sx={merge<BetterSystemStyleObject>(
+        {
+          fontSize: {
+            large: '2rem',
+            medium: '1.25rem',
+            subtitle: '1.25rem',
```

Comment:
The typography looks identical for `medium` and `subtitle`. Is this intentional?